### ISSUE TITLE
The argument "network" is required, but no definition was found.

### ISSUE THREAD

**Issue by DeLoWaN** [terraform-google-modules/terraform-google-sql-db#417]
### TL;DR

On private-service-access, the use of data resource to get the network id may end up in inapplicable configuration, for example when there is a change in the network name.

### Expected behavior

Changes are applied

### Observed behavior

```
| Error: Missing required argument
│ 
│   with *****.google_service_networking_connection.private_service_access,
│   on ***** modules/private_service_access/main.tf line 43, in resource "google_service_networking_connection" "private_service_access":
│   43:   network                 = data.google_compute_network.main.self_link
│ 
│ The argument "network" is required, but no definition was found.

### Terraform Configuration

```hcl
Not needed
```

### Terraform Version

```sh
Terraform v1.3.7
on darwin_arm64
+ provider registry.terraform.io/hashicorp/google v4.50.0
+ provider registry.terraform.io/hashicorp/google-beta v4.50.0
+ provider registry.terraform.io/hashicorp/null v3.2.1
+ provider registry.terraform.io/hashicorp/random v3.4.3
+ provider registry.terraform.io/hashicorp/time v0.9.1
```
```


### Additional information

I already have a working configuration. I wanted then to change the network name. It ended up with an inapplication configuration, I cannot finish the `terraform plan`.

Why ? Since the data resource used in network-private-access is looking for a network that is not created yet (since its name is going to change).

This is fairly easy reproductible. Just create a network-private-access and try change the name of the network.

The solution would be not to use a data resource to fetch the network and instead passing directly to the module the network id or self_link. This way, the data would not be looking for anything. But, it means that it would be a major change with no backward compatibility.
